dotCMS/src/main/java/com/dotmarketing/portlets/languagesmanager/business/LanguageCacheImpl.java

```diff
@@ -199,35 +205,121 @@ private void removeFromCache(Language language) {
+		if (perLangCache == null) {
+			return;
+		}
+		@SuppressWarnings("unchecked")
+		final ConcurrentMap<String,List<LanguageVariable>> langVarCache = (ConcurrentMap<String,List<LanguageVariable>>) perLangCache;
+		final String languageIdStr = String.valueOf(languageId);
+		langVarCache.remove(languageIdStr);
+		Logger.debug(this, "Language Variables for language with id: " + languageId + " have been removed from cache.");
+	}
+
+	/**
+	 * This method is used to clear the cache for all language variables
+	 */
+	public void clearVariables(){
+		final String group = getSecondaryGroup();
+		final DotCacheAdministrator cache = CacheLocator.getCacheAdministrator();
+		cache.flushGroup(group);
+	}
+
+
+    /**
+	 * This method is used to store a list of LanguageVariables in cache
+	 * @param languageId the language id to store the list of LanguageVariables for
+	 * @param vars the list of LanguageVariables to store
+	 */
+	@Override
+	public void putVars(final long languageId, final List<LanguageVariable> vars){
+		final String languageIdStr = String.valueOf(languageId);
+		final String group = getSecondaryGroup();
+		final DotCacheAdministrator cache = CacheLocator.getCacheAdministrator();
+		Object perLangCache = cache.getNoThrow(LANG_VARIABLES_CACHE, group);
+		if (perLangCache == null) {
+			// A map of LanguageVariables is stored per language
+			cache.put(LANG_VARIABLES_CACHE, new ConcurrentHashMap<String,List<LanguageVariable>>(), group);
+			perLangCache = cache.getNoThrow(LANG_VARIABLES_CACHE, group);
+		}
+		@SuppressWarnings("unchecked")
+		final ConcurrentMap<String,List<LanguageVariable>> langVarCache = (ConcurrentMap<String,List<LanguageVariable>>) perLangCache;
+		//Now we use the crafted (specific) key to store the list of LanguageVariables
+		//So that if we want to invalidate only the list of LanguageVariables for a specific language it is possible
+		langVarCache.put(languageIdStr, vars);
+		cache.put(languageIdStr, langVarCache, group);
+	}
+
+	/**
+	 * This method is used to retrieve a list of LanguageVariables from cache
+	 * @param languageId the language id
+	 * @return a list of LanguageVariables
+	 */
+	public List<LanguageVariable> getVars(final long languageId){
+		final String group = getSecondaryGroup();
+		final DotCacheAdministrator cache = CacheLocator.getCacheAdministrator();
+		final String languageIdStr = String.valueOf(languageId);
+		// A map of LanguageVariables is stored per language
+		Object  perLangCache = cache.getNoThrow(LANG_VARIABLES_CACHE, group);
+		if (perLangCache == null) {
+			return List.of();
+		}
+		@SuppressWarnings("unchecked")
+		final ConcurrentMap<String,List<LanguageVariable>> langVarCache = (ConcurrentMap<String,List<LanguageVariable>>) perLangCache;
+		//Now we use the crafted (specific) key to access the list of LanguageVariables
+		final List<LanguageVariable> variables = langVarCache.get(languageIdStr);
+		if (variables == null) {
+			return List.of();
```

Comment:
I think we should return `null` instead of an empty list, empty is a valid state.

Reply:
same as stated above